src/pybamm/solvers/summary_variable.py

```diff
@@ -0,0 +1,182 @@
+#
+# Summary Variable class
+#
+from __future__ import annotations
+import pybamm
+import numpy as np
+from typing import Any
+
+
+class SummaryVariables:
+    """
+    Class for managing and calculating summary variables from a PyBaMM solution.
+    Summary variables are only calculated when simulations are run with PyBaMM
+    Experiments.
+
+    Parameters
+    ----------
+    solution : :class:`pybamm.Solution`
+        The solution object to be used for creating the processed variables.
+    cycle_summary_variables : list[pybamm.SummaryVariables], optional
+        A list of cycle summary variables.
+    esoh_solver : :class:`pybamm.lithium_ion.ElectrodeSOHSolver`, optional
+        Solver for electrode state-of-health (eSOH) calculations.
+    user_inputs : dict, optional
+        Additional user inputs for calculations.
+    """
+
+    def __init__(
+        self,
+        solution: pybamm.Solution,
+        cycle_summary_variables: list[SummaryVariables] | None = None,
+        esoh_solver: pybamm.lithium_ion.ElectrodeSOHSolver | None = None,
+        user_inputs: dict[str, Any] | None = None,
+    ):
+        self.user_inputs = user_inputs or {}
+        self.esoh_solver = esoh_solver
+        self._variables = {}  # Store computed variables
+        self.cycle_number = None
+
+        model = solution.all_models[0]
+        self._possible_variables = model.summary_variables  # minus esoh variables
+        self._esoh_variables = None  # Store eSOH variable names
+
+        # Flag if eSOH calculations are needed
+        self.calc_esoh = (
+            self.esoh_solver is not None
+            and isinstance(model, pybamm.lithium_ion.BaseModel)
+            and model.options.electrode_types["negative"] == "porous"
+            and "Negative electrode capacity [A.h]" in model.variables
+            and "Positive electrode capacity [A.h]" in model.variables
+        )
+
+        # Initialize based on cycle information
+        if cycle_summary_variables:
+            self._initialize_for_cycles(cycle_summary_variables)
+        else:
+            self.first_state = solution.first_state
+            self.last_state = solution.last_state
+            self.cycles = None
+
+    def _initialize_for_cycles(self, cycle_summary_variables: list[SummaryVariables]):
+        """Initialize attributes for when multiple cycles are provided."""
+        self.first_state = None
+        self.last_state = None
+        self.cycles = cycle_summary_variables
+        self.cycle_number = np.arange(1, len(self.cycles) + 1)
```

Comment:
I can see that at least `cycle_number` is used outside this class, can you please document the attributes that are meant to be public in the docstring?